src/tr2/game/objects/general/lift.c

```diff
@@ -77,43 +89,47 @@ static void M_FloorCeiling(
     const bool lara_in_shaft =
         (lara_tile.x == lift_tile.x || lara_tile.x + dx == lift_tile.x) &&
         (lara_tile.z == lift_tile.z || lara_tile.z + dz == lift_tile.z);
-    // clang-format on
 
+    const int32_t lift_bottom = item->pos.y + STEP_L;
     const int32_t lift_floor = item->pos.y;
-    const int32_t lift_ceiling = item->pos.y - LIFT_HEIGHT;
+    const int32_t lift_ceiling = item->pos.y - LIFT_HEIGHT + STEP_L;
+    const int32_t lift_top = item->pos.y - LIFT_HEIGHT;
+
+    const bool lara_inside_lift = (lara_item->pos.y < lift_bottom) &&
+                                  (lara_item->pos.y > lift_ceiling);
+    // clang-format on
```

Comment:
One additional cosmetic change I did make was clean up these names which I hope people agree are easier to understand. Here's a picture to demonstrate:

<img width="3840" height="2160" alt="image" src="https://github.com/user-attachments/assets/b095d1d2-b3db-4b5c-a178-a5b19609288e" />